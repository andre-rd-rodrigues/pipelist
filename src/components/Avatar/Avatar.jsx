import React, { useState } from "react";
import axios from "api/pipelistAxios";
import FeatherIcon from "feather-icons-react";
import styles from "./avatar.module.scss";

const Avatar = ({ src, size = 4, editPicture }) => {
  const [uploadedPicture, setUploadedPicture] = useState(undefined);

  //Http
  const postPicture = async (uploadedFile) => {
    let formData = new FormData();
    formData.append("file", uploadedFile);

    return axios
      .post(`/persons/${editPicture}/picture`, formData, {
        headers: {
          "Content-Type": "multipart/form-data"
        }
      })
      .then(() => {})
      .catch(() => {
        return;
      });
  };

  const handleUpload = (input) => {
    setUploadedPicture(URL.createObjectURL(input.target.files[0]));
    return postPicture(input.target.files[0]);
  };

  return (
    <div
      className={styles.avatar}
      id="app_avatar"
      style={{
        width: `${size}rem`,
        height: `${size}rem`,
        backgroundImage: `url(${
          src ||
          uploadedPicture ||
          process.env.PUBLIC_URL + "/images/avatar.png"
        })`
      }}
    >
      {editPicture && (
        <div role="button">
          <FeatherIcon icon="camera" color="white" size={size * 6} />
          <input type="file" name="file" onChange={handleUpload} />
        </div>
      )}
    </div>
  );
};

export default Avatar;
