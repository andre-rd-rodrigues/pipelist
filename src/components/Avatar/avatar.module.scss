@import "styles/mixins";

.avatar {
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
  position: relative;
  overflow: hidden;

  &:hover {
    & div[role="button"] {
      opacity: 1;
    }
  }

  & div[role="button"] {
    height: 100%;
    width: 100%;
    background-color: rgb(215, 215, 215);
    display: flex;
    justify-content: center;
    align-items: center;
    opacity: 0;
    transition: 0.2s ease;
  }
}
