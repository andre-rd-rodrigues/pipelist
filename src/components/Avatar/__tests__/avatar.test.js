import { render, screen } from "@testing-library/react";
import { LoadingProvider } from "context/loading-context";
import reactTestRenderer from "react-test-renderer";
import Avatar from "../Avatar";

let providerProps = {
  loading: false,
  setLoading: jest.fn()
};

const MockProvider = ({ children }) => (
  <LoadingProvider value={providerProps}>{children}</LoadingProvider>
);

describe("Avatar", () => {
  it("renders correctly", () => {
    const wrapper = reactTestRenderer.create(
      <MockProvider>
        <Avatar />
      </MockProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders upload component from profileId", () => {
    render(
      <MockProvider>
        <Avatar profileId={1} />
      </MockProvider>
    );

    expect(screen.getByRole("button")).toBeInTheDocument();
  });
});
