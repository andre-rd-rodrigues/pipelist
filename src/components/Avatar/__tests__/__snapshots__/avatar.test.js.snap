// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  className="avatar"
  id="app_avatar"
  style={
    Object {
      "backgroundImage": "url(/images/avatar.png)",
      "height": "4rem",
      "width": "4rem",
    }
  }
/>
`;
